specs-go/config.go

```diff
@@ -469,3 +479,77 @@ type Syscall struct {
 	Action Action `json:"action"`
 	Args   []Arg  `json:"args,omitempty"`
 }
+
+// Windows contains platform specific configuration for Windows based containers.
+type Windows struct {
+	// Resources contains constraints for containers.
+	Resources *WindowsResources `json:"resources,omitempty"`
+	// Networking contains the network connection settings for containers.
+	Networking *Networking `json:"networking,omitempty"`
+	// FirstStart is an optimization on first boot of a container.
+	FirstStart bool `json:"first_start,omitempty"`
+	// LayerFolder is the path to the current layer folder.
+	LayerFolder string `json:"layer_folder,omitempty"`
+	// Layer contains the filepaths of the parent layers.
+	LayerPaths []string `json:"layer_paths,omitempty"`
+	// Settings relating to Hyper-V containers, omitted otherwise.
+	HvRuntime *HvRuntime `json:"hv_runtime,omitempty"`
+}
+
+// WindowsResources has container runtime resource constraints for Windows based containers.
+type WindowsResources struct {
+	// Network resource configuration.
+	Network *WindowsNetwork `json:"network,omitempty"`
+	// Memory resource configuration.
+	Memory *WindowsMemory `json:"memory,omitempty"`
+	// CPU resource configuration.
+	CPU *WindowsCPU `json:"cpu,omitempty"`
+	// Storage resource configuration.
+	Storage *WindowsStorage `json:"storage,omitempty"`
+}
+
+// WindowsNetwork has network runtime resource constraings for Windows based containers.
+type WindowsNetwork struct {
+	// Bandwidth is the maximum egress bandwidth in bytes per second.
+	Bandwidth *uint64 `json:"bandwidth,omitempty"`
+}
+
+// WindowsMemory has memory runtime resource constraints for Windows based containers.
+type WindowsMemory struct {
+	// Memory limit (in bytes).
+	Limit *int64 `json:"limit,omitempty"`
```

Comment:
Is this intentionally signed?  If so, please explain what negative values mean when you land the Markdown docs ;).